redshift/resource_redshift_user.go

```diff
@@ -253,17 +253,31 @@ func resourceRedshiftUserRead(db *DBConnection, d *schema.ResourceData) error {
 }
 
 func resourceRedshiftUserReadImpl(db *DBConnection, d *schema.ResourceData) error {
-	var userName, userValidUntil, userConnLimit, userSyslogAccess string
+	var userName, userValidUntil, userConnLimit, userSyslogAccess, userInfoTable string
 	var userSuperuser, userCreateDB bool
-
-	columns := []string{
-		"usename",
-		"usecreatedb",
-		"usesuper",
-		"syslogaccess",
-		`COALESCE(useconnlimit::TEXT, 'UNLIMITED')`,
+	var columns []string
+
+	if db.client.config.IsServerless {
+		columns = []string{
+			"usename",
+			"usecreatedb",
+			"usesuper",
+			"'RESTRICTED'",
+			"'UNLIMITED'",
+		}
+		userInfoTable = "pg_user"
+	} else {
+		columns = []string{
+			"usename",
+			"usecreatedb",
+			"usesuper",
+			"syslogaccess",
+			`COALESCE(useconnlimit::TEXT, 'UNLIMITED')`,
+		}
+		userInfoTable = "svl_user_info"
```

Comment:
We can avoid forking between serverless and provisioned if we use the `svv_user_info` view which is available on both instance types. The column names need to be updated to reflect the change (e.g. `usename` becomes `user_name`).